File: BackEnd/Praza-Shop/src/main/java/org/example/prazashop/model/entity/Usuario.java
package org.example.prazashop.model.entity;

import jakarta.persistence.*;
import lombok.*;
import org.example.prazashop.auth.repository.Token;
import org.example.prazashop.model.TipoUsuario;

import java.util.List;

/**
 * The type Usuario.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "usuario")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String contrasinal;

    private String telefono;

    @Enumerated(EnumType.STRING)
    private TipoUsuario tipoUsuario;
    @OneToMany(mappedBy = "usuario", fetch = FetchType.LAZY)
    private List<Token> tokens;

}

File: FrontEnd/praza_shop/lib/models/usuario.dart
class Usuario {
  final int? id;
  final String nome;
  final String? email;
  final String? telefono;
  final String? tipoUsuario;

  Usuario({this.id, required this.nome, this.email, this.telefono, this.tipoUsuario});

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json['id'] is int ? json['id'] : (json['id'] != null ? int.tryParse('${json['id']}') : null),
        nome: json['nome'] ?? json['name'] ?? '',
        email: json['email'],
        telefono: json['telefono'],
        tipoUsuario: json['tipoUsuario'] ?? json['role'],
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'nome': nome,
        if (email != null) 'email': email,
        if (telefono != null) 'telefono': telefono,
        if (tipoUsuario != null) 'tipoUsuario': tipoUsuario,
      };
}
